### PR TITLE
Implemented getAllCredentials method for etcd credentials store

### PR DESCRIPTION
**Pull request by eamansour**
## Why?
See https://github.com/galasa-dev/projectmanagement/issues/1467

## Changes
- Added `getAllCredentials()` implementation to get all credentials in the `secure` namespace from etcd

**Note: Builds for this PR will fail until https://github.com/galasa-dev/framework/pull/662 is delivered**

### CODE REVIEW

File: galasa-extensions-parent/dev.galasa.cps.etcd/src/main/java/dev/galasa/cps/etcd/internal/Etcd3CredentialsStore.java [hunk 1]
```diff
@@ -136,10 +135,80 @@ public void setCredentials(String credentialsId, ICredentials credentials) throw
...
-            deletePrefix("secure.credentials." + credentialsId);
+            deletePrefix(CREDS_PROPERTY_PREFIX + "." + credentialsId);
         } catch (InterruptedException | ExecutionException e) {
             Thread.currentThread().interrupt();
             throw new CredentialsException("Failed to delete credentials", e);
         }
     }
+
+    @Override
+    public Map<String, ICredentials> getAllCredentials() throws CredentialsException {
```
Comment (techcobweb) [inline galasa-extensions-parent/dev.galasa.cps.etcd/src/main/java/dev/galasa/cps/etcd/internal/Etcd3CredentialsStore.java:146]: getting all the credentials seems a very scary thing to be doing no ? Why do we need to do this ?

File: galasa-extensions-parent/dev.galasa.cps.etcd/src/main/java/dev/galasa/cps/etcd/internal/Etcd3CredentialsStore.java [hunk 2]
```diff
@@ -136,10 +135,80 @@ public void setCredentials(String credentialsId, ICredentials credentials) throw
...
+            for (String id : credentialIds) {
+                Map<String, String> idProperties = credentialsEntries.stream()
+                    .filter(entry -> entry.getKey().contains("." + id + "."))
+                    .collect(Collectors.toMap(Entry::getKey, Entry::getValue));
+
+                String username = idProperties.get(CREDS_PROPERTY_PREFIX + "." + id + ".username");
+                String password = idProperties.get(CREDS_PROPERTY_PREFIX + "." + id + ".password");
+                String token = idProperties.get(CREDS_PROPERTY_PREFIX + "." + id + ".token");
+
+                ICredentials convertedCredentials = convertValuesIntoCredentials(username, password, token);
```
Comment (techcobweb) [inline galasa-extensions-parent/dev.galasa.cps.etcd/src/main/java/dev/galasa/cps/etcd/internal/Etcd3CredentialsStore.java:171]: this and the 3 lines above are duplicates ? Do in a separate method ? ?

File: galasa-extensions-parent/dev.galasa.cps.etcd/src/main/java/dev/galasa/cps/etcd/internal/Etcd3CredentialsStore.java (hunk 1 above)
Comment (eamansour) [reply galasa-extensions-parent/dev.galasa.cps.etcd/src/main/java/dev/galasa/cps/etcd/internal/Etcd3CredentialsStore.java:146]: It was a requirement documented in the story - https://github.com/galasa-dev/projectmanagement/issues/1467

This is used by the `GET /secrets` endpoint, which is used in `galasactl secrets get` - should this only return the keys and metadata?

**Review by techcobweb — APPROVED** [review]
approved subject to comments being sorted out.

**Pull request merged**